src/api/routes/awala.routes.ts

```diff
@@ -0,0 +1,96 @@
+import type { RouteOptions } from 'fastify';
+import Ajv from 'ajv';
+import addFormats from 'ajv-formats';
+import type { JSONSchema, FromSchema } from 'json-schema-to-ts';
+
+import { HTTP_STATUS_CODES } from '../../utilities/http.js';
+import type { PluginDone } from '../../utilities/fastify/PluginDone.js';
+import type { FastifyTypedInstance } from '../../utilities/fastify/FastifyTypedInstance.js';
+import { MEMBER_BUNDLE_REQUEST_SCHEMA } from '../../schemas/awala.schema.js';
+import { AwalaProblemType } from '../../AwalaProblemType.js';
+import { createMemberBundleRequest } from '../../awala.js';
+import type { ServiceOptions } from '../../serviceTypes.js';
+
+const ajv = addFormats(new Ajv());
+
+type ValidationResult<Schema extends JSONSchema> = FromSchema<Schema> | string;
+
+function validateMessage<Schema extends JSONSchema>(
+  value: unknown,
+  schema: Schema,
+): ValidationResult<Schema> {
+  if (!ajv.validate(schema, value)) {
+    return ajv.errorsText(ajv.errors);
+  }
+
+  return value as ValidationResult<Schema>;
+}
+
+async function processMemberBundleRequest(
+  data: unknown,
+  options: ServiceOptions,
+): Promise<AwalaProblemType | undefined> {
+  const validationResult = validateMessage(data, MEMBER_BUNDLE_REQUEST_SCHEMA);
+  if (typeof validationResult === 'string') {
+    options.logger.info(data, validationResult);
+    return AwalaProblemType.MALFORMED_AWALA_MESSAGE_BODY;
+  }
+
+  await createMemberBundleRequest(validationResult, options);
+  return undefined;
+}
+
+enum AwalaRequestMessageType {
+  MEMBER_BUNDLE_REQUEST = 'application/vnd.veraid.member-bundle-request',
+}
+
+const awalaEventToProcessor: {
+  [key in AwalaRequestMessageType]: (
+    data: unknown,
+    options: ServiceOptions,
+  ) => Promise<AwalaProblemType | undefined>;
+} = {
+  [AwalaRequestMessageType.MEMBER_BUNDLE_REQUEST]: processMemberBundleRequest,
+};
+const awalaRequestMessageTypeList: AwalaRequestMessageType[] =
+  Object.values(AwalaRequestMessageType);
+
+export default function registerRoutes(
+  fastify: FastifyTypedInstance,
+  _opts: RouteOptions,
+  done: PluginDone,
+): void {
+  fastify.addContentTypeParser(
+    awalaRequestMessageTypeList,
+    {
+      parseAs: 'string',
+    },
+    fastify.getDefaultJsonParser('ignore', 'ignore'),
+  );
+
+  fastify.route({
+    method: ['POST'],
+    url: '/awala',
+
+    async handler(request, reply): Promise<void> {
+      const contentType = awalaRequestMessageTypeList.find(
+        (messageType) => messageType === request.headers['content-type'],
+      );
+
+      const serviceOptions = {
+        logger: this.log,
+        dbConnection: this.mongoose,
+      };
+
+      const result = await awalaEventToProcessor[contentType!](request.body, serviceOptions);
```

Comment:
This statement could be a litter simpler, which would also make it easier to debug:

```suggestion
      const processor = awalaEventToProcessor[contentType!];
      const result = await processor(request.body, serviceOptions);
```